packages/salesforcedx-vscode-core/src/commands/templates/apexGenerateClass.ts

```diff
@@ -49,9 +51,18 @@ export const getParamGatherers = () => {
   };
 };
 
-export const apexGenerateClass = async () => {
+// if called from a file's context menu, will deliver the clicked file URI,
+// ignoring an additional arg that is array of selected
+// if called from the command pallet args will be empty
+export const apexGenerateClass = async (sourceUri?: vscode.Uri) => {
   const gatherers = getParamGatherers();
 
+  if (sourceUri) {
```

Comment:
This is the only necessary change, this block alone enabled the feature. Everything else is actually to improve testing and typings.